### PR TITLE
Closes #1: Fix caching of rdds that was loaded with SparkAvroLoader

### LINKED ISSUE #1: Fix avro rdd loader
When using cache() on rdd that was loaded with SparkAvroLoader we get strange results.
Spark sometimes show warning
`Caching NewHadoopRDDs as deserialized objects usually leads to undesired behavior because Hadoop's RecordReader reuses the same Writable object`
which describes cause of this error

### PR DESCRIPTION
**Pull request by madryk**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -162,6 +163,79 @@
...
+                                	<action>
+                                		<ignore></ignore>
+                                	</action>
+                                </pluginExecution>
+                            </pluginExecutions>
+                        </lifecycleMappingMetadata>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
```
Comment (lukdumi) [inline pom.xml]: It's better to only add this to the eclipse configuration, not the pom.

File: src/main/java/pl/edu/icm/sparkutils/avro/SparkAvroLoader.java [hunk 2]
```diff
@@ -49,8 +50,10 @@ private SparkAvroLoader() {
         @SuppressWarnings("unchecked")
         JavaPairRDD<AvroKey<T>, NullWritable> inputRecords = (JavaPairRDD<AvroKey<T>, NullWritable>)
                 sc.newAPIHadoopFile(avroDatastorePath, AvroKeyInputFormat.class, avroRecordClass, NullWritable.class, job.getConfiguration());
+        
+        
 
-        JavaRDD<T> input = inputRecords.map(tuple -> tuple._1.datum());
+        JavaRDD<T> input = inputRecords.map(tuple -> AvroUtils.cloneAvroRecord(tuple._1.datum()));
```
Comment (lukdumi) [inline src/main/java/pl/edu/icm/sparkutils/avro/SparkAvroLoader.java:59]: A very clever move ;-)
But it needs explanation, a comment here above the line and above the method.

File: pom.xml (hunk 1 above)
Comment (madryk) [inline pom.xml]: ok. I will remove it from the pom

File: src/main/java/pl/edu/icm/sparkutils/avro/SparkAvroLoader.java (hunk 2 above)
Comment (madryk) [inline src/main/java/pl/edu/icm/sparkutils/avro/SparkAvroLoader.java:59]: I added comment for this line.
Not sure what to write above the method though

**Pull request closed without merging**